### ISSUE TITLE
Moving TrafficManaget to this repo

### ISSUE THREAD

**Issue by EtienneBruines** [EngoEngine/engo#262]
Main issue: we want to add [TrafficManager]() to the travis configuration, to ensure it builds. (Note that this will not ensure the "code snippets" at engo.io will build, but at least the source code is up-to-date. ) 

We could do this by moving the branches at TrafficManager into separate directories at this repo. 
- `demos/trafficmanager/01-hello-world`
- `demos/trafficmanager/02-first-system`
- etc.

I originally thought it would result in ugly import paths (and it does!), but I'd rather have it build than have a nice import path (for internal demo-packages). 

---

The alternative would be to manually `git clone` the TrafficManager repo, `git checkout` the correct branch, check if it `go build`s, and repeat that process for each and every branch -- and defining this in the `.travis.yml` file. 

---

Discuss :smile:

**Comment by otraore** [comment]
@EtienneBruines no need to loop the branches, only need to check the master branch when making sure it builds.

**Comment by paked** [comment]
Isn't there a way to trigger other travis builds while in a travis build?
If that is a thing we could setup a travis.yml file in the TrafficManager
repo itself.

**Comment by EtienneBruines** [comment]
@paked That would be awesome, but I haven't found anything in the documentation that would indicate such a feature. 

@otraore So we'd just "trust" that whenever we have to make changes to `master`, other branches are being updated as well? (There's 1 branch per tutorial-post, and I'd love for each of those to be kept up-to-date).

**Comment by otraore** [comment]
@EtienneBruines Why are tutorial-posts per branch?

**Comment by otraore** [comment]
Also https://docs.travis-ci.com/user/triggering-builds ?

**Comment by EtienneBruines** [comment]
@otraore They are per-branch, because the tutorial posts work incrementally. So each new post, adds something new -- possibly changing / removing what was previously there. 

So in order to prevent people from seeing a source directory with 10 files and thousands of lines of code, it's just limited to what was explained in that specific post. So each branch handles just that what is handled by the tutorial post.

**Comment by otraore** [comment]
SGTM, I've actually seen that a couple of times with tutorial like repos. Alright so moving them to master is not an option then

**Comment by EtienneBruines** [comment]
Well, we could move each branch to its own folder. That would "flatten" the repository, and make it easier to move it to this one.

**Comment by paked** [comment]
I kind of want to keep traffic manager in its own repo for now... What if we did something like:

```bash
# WARNING: JANK BASH AHEAD

go get github.com/EngoEngine/TrafficManager
cd $GOPATH/src/github.com/EngoEngine/TrafficManager
for branch in $(git branch | grep -v "*"); do git checkout $branch && go test -v; done
```

**Comment by otraore** [comment]
I don't tthink go get git clones a repo, and even if it did not all the branches show up automatically, I'm going give this a try with automatically triggering a build

**Comment by paked** [comment]
@otraore `go get` clones the repo for me?

**Comment by EtienneBruines** [comment]
@paked @otraore more specifically, it `git clone`s indeed - checking out the `master` branch by default. And there's no way to change this (except for `git checkout` afterwards)

**Comment by otraore** [comment]
I vote we keep TrafficManager in it's own repo.

**Comment by paked** [comment]
Me too. I'd really like to setup build triggering though.

Or even if we just cloned it and ran the tests.

**Comment by Noofbiz** [comment]
I'm going to go with just cloning it and attempting to build each branch without moving it into the demos, unless anyone objects.

**Comment by Noofbiz** [comment]
Closed by #543
